src/parser.cpp

```diff
@@ -8,15 +8,29 @@
 //
 
 #include <boost/http_proto/parser.hpp>
+
 #include <boost/http_proto/context.hpp>
 #include <boost/http_proto/error.hpp>
+#include <boost/http_proto/rfc/detail/rules.hpp>
```

Comment:
is rules.hpp public?

Reply:
This one is because headers like list_rule rely on it.

Reply:
could it be a private common header?

Reply:
I think so, yeah.
We have a couple of public rule types in the rfc/ folder, namely list_rule and upgrade_rule
upgrade_rule isn't used by any other file except the project's convenience header. 
I think we should be able to move all of these into `src/` no problem.